packages/core/strapi/lib/types/factories.d.ts

```diff
@@ -49,12 +61,42 @@ type ControllerCallback<T extends GenericController = GenericController> = (para
 }) => T;
 type ServiceCallback<T extends GenericService = GenericService> = (params: { strapi: Strapi }) => T;
 
-export function createCoreRouter(uid: string, cfg?: RouterConfig = {}): () => Router;
-export function createCoreController<T extends GenericController = GenericController>(
-  uid: string,
-  cfg?: ControllerCallback<T> | T = {}
-): () => T & Controller;
-export function createCoreService<T extends GenericService = GenericService>(
-  uid: string,
-  cfg?: ServiceCallback<T> | T = {}
-): () => T;
+export function createCoreRouter<T extends SchemaUID>(
```

Comment:
So, this is working and just adding to what was already there, so I wouldn't address it in this PR, but I'm a little confused why we're exporting a function from a .d.ts.

Is it because factory.js itself is a .js file and so the function definition has to be defined like this? It seems like there has to be a better way.

Reply:
It's missing a `declare` keyword to fit the .d.ts but other than that, this is correct :slightly_smiling_face:

Reply:
`export declare function foo(): 'bar';`

Reply:
Should I add it?

Reply:
Ahhh, ok, yes, that makes sense